### ISSUE TITLE
Installation of IPFS on mirror

### ISSUE THREAD

**Issue by mitra42** [internetarchive/dweb-mirror#22]
See #11 

Notes from @parkan 
- to use filestore just pass `&nocopy=true`w
- install from these binaries, needs a bit of platform detection, will look for an automagic script https://dist.ipfs.io/#go-ipfs
- filestore is an experimental feature so need to run `ipfs config --json Experimental.FilestoreEnabled true` once after installing & starting daemon

**Comment by mitra42** [comment]
Installation is done, (`install.sh` and `ipfs_container_daemon.sh` in the `#ipfs` branch) but still need to figure out how to autostart at same time as mirrorHttp.

**Comment by parkan** [comment]
might be good to use supervisord's groups for this: http://supervisord.org/configuration.html#group-x-section-settings

(will also help with auto-restarts etc)

**Comment by mitra42** [comment]
Yes - that's what we do on the production servers at the IA, there are two challenges - one  addressed in https://github.com/ipfs/go-ipfs/issues/5983 was related to knowing whether the daemon was already started. I've now got the workaround I posted there, but its a bit of a kludge! 

Second issue is that the platforms we want this to go on don't necessarily have supervisord so for example this isn't going to work on a mac. I think this is my problem, not IPFS's though it would be easier I think if the suggestion I made in 5983 happened.

**Comment by parkan** [comment]
it sounds like you want similar functionality (detach) to start-stop-daemon? honestly this is done very inconsistently across different servers and I wouldn't expect super nice behavior in userland without using a harness like supervisord (not really the right place to do daemon wrangling)

checking $? of `ipfs swarm addrs` is quite sound, it ensures that the startup got to the point of binding addresses, which is something that you want

**Comment by mitra42** [comment]
I agree with you, I'd rather see supervisord, but on some of these platforms they are seriously stripped down (such as the Rachel3+ wifi/storage appliance) and on others (e.g. Mac) its not supervisord and I haven't figured out (just not gotten to it yet) how to insert a daemon start on a Mac. 

I've not used start-stop-daemon (more familiar with supervisord). I think having a supported function in IPFS would be good (e.g. appropriate flags to `ipfs daemon` rather than depending on the return code behaviour of `ipfs swarm addrs`) since my instinct is that that behavior of `swarm addrs` could change, which is what I'll use for now.
